pymongoexplain/__main__.py

```diff
@@ -13,48 +13,35 @@
 # limitations under the License.
 
 
-from pymongo import monitoring, MongoClient
 from pymongo.collection import Collection
+from .explainable_collection import ExplainCollection
+
 import sys
-from bson.son import SON
+import logging
 
-class CommandLogger(monitoring.CommandListener):
-    def __init__(self):
-        self.payloads = []
+FORMAT = '%(asctime)s %(levelname)s %(module)s %(message)s'
+logging.basicConfig(format=FORMAT, level=logging.INFO)
 
-    def started(self, event):
-        print("started")
-        self.payloads.append(event.command)
+old_function_names = ["update_one", "replace_one", "update_many", "delete_one",
+                      "delete_many", "aggregate", "watch", "find", "find_one",
+                      "find_one_and_delete", "find_one_and_replace",
+                      "find_one_and_update", "count_documents",
+                      "estimated_document_count", "distinct"]
+old_functions = [getattr(Collection, i) for i in old_function_names]
 
-    def succeeded(self, event):
-        pass
 
-    def failed(self, event):
-        pass
+def make_func(old_func, old_func_name):
+    def new_func(self: Collection, *args, **kwargs):
+        logging.info(getattr(ExplainCollection(self), old_func_name)(*args,
```

Comment:
Could you add some context to the long line? Maybe something as simple as "update_one explained: {...}"?

Reply:
Context on what it does? Or just an example of the output?

Reply:
More context in the log line itself. Right now we just log the response:
```
2020-07-16 18:02:14,209 INFO __main__ {'queryPlanner': ...}
```
It would be nicer to include the method name being explained so that the user know what the output is. Like this:
```
2020-07-16 18:02:14,209 INFO __main__ update_one explain response: {'queryPlanner': ...}
2020-07-16 18:02:14,209 INFO __main__ find_one explain response: {'queryPlanner': ...}
```

Reply:
oh I see what you mean. Will do